sparkly/functions.py

```diff
@@ -153,3 +153,26 @@ def _execute_case(accumulator, case):
     result = reduce(_execute_case, cases, F).otherwise(default)
 
     return result
+
+
+def argmax(field, by, condition=None, nullable=False):
+    """Get t he `field` value that maximizes columns in `by`
+
+    Arguments:
+        field (string): the field to return (and the alias to use)
```

Comment:
do `field` and `by` need to be strings or will this support other column expressions, like the following?

`argmax(F.coalesce(F.col('a'), F.col('b')), by=[F.col('a') + F.col('d')])`